h2o-genmodel/src/main/java/hex/genmodel/MojoModel.java

```diff
@@ -71,5 +76,38 @@ public static MojoModel load(MojoReaderBackend mojoReader) throws IOException {
   protected MojoModel(String[] columns, String[][] domains, String responseColumn) {
     super(columns, domains, responseColumn);
   }
+
+  // RowConverterFactory
+
+  @Override
+  public RowToRawDataConverter _makeRowConverter(CategoricalEncoding categoricalEncoding,
+                                                 ErrorConsumer errorConsumer,
+                                                 Config config) {
+    if (_preprocessors != null) {
+      RowToRawDataConverter[] converters = new RowToRawDataConverter[_preprocessors.length+1];
+      int i = 0;
+      GenModel preprocessedModel = this;
+      for (MojoPreprocessor preprocessor : _preprocessors) {
+        MojoPreprocessor.ModelProcessor processor = preprocessor.makeProcessor(preprocessedModel);
+        converters[i] = processor.makeRowConverter(errorConsumer, config);
+        preprocessedModel = processor.getProcessedModel();
+        i++;
+      }
+      converters[i] = new CategoricalEncodingAsModelProcessor(preprocessedModel, this, categoricalEncoding).makeRowConverter(errorConsumer, config);
+      return new CompositeRowToRawDataConverter<>(converters);
+    }
+    
+    //FIXME: currently some categoricalEncoding that create new columns (binary, onehot...) don't put the non-predictors columns (e.g. fold) to the end (wrong offset).
```

Comment:
this is actually PUBDEV-7775, won't fix in this PR as it's not TE-related and doesn't impact the goal of this  integration for AutoML (AutoML uses only AUTO categorical encoding).

Reply:
removing this FIXME: using its content in PUBDEV-7775 description.